sdk/core/Azure.Core.Experimental/src/HttpPipelineExtensions.cs

```diff
@@ -0,0 +1,38 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+using Azure.Core.Pipeline;
+
+namespace Azure.Core.Experimental
+{
+    /// <summary>
+    /// Extensions to HttpPipeline to support RequestOptions.
+    /// </summary>
+    public static class HttpPipelineExtensions
```

Comment:
Can we call this `AzureCoreExtensions` so it'll eventually merge cleanly into https://apiview.dev/Assemblies/Review/77bdbcef49a3465fa96fac3fb745bf57#Azure.AzureCoreExtensions?  I like to avoid having multiple extensions classes per assembly where possible.  We can do this when merging into Core as well if preferable.

Reply:
It'll be a method directly on HttpPipeline when we move it to Core.